packages/devtools/src/composite.ts

```diff
@@ -57,14 +65,98 @@ function assertModelsHaveCommits(
   }
 }
 
-function assertSupportedModelController(model: Model): void {
-  if (!model.metadata.controller.startsWith('did:key:')) {
+function assertSupportedWriteModelController(model: Model, ceramic: CeramicApi): void {
+  const unsupported = `Unsupported model controller ${model.metadata.controller}`
+  if (model.metadata.controller.startsWith('did:pkh:')){
+    if (ceramic.context.did == null) {
+      throw new Error(
+        `${unsupported}, did missing from ceramic context`
+      )
+    }
+    if (!ceramic.context.did.hasCapability()) {
+      throw new Error(
+        `${unsupported}, only did:pkh with CACAO is supported`
+      )
+    }
+    const cacao: Cacao = ceramic.context.did.capability(); 
+    if (cacao.p.exp == null) {
+      throw new Error(
+        `${unsupported}, only did:pkh CACAO without expiry is supported`
+      )
+    }
+    const hasModelResource = cacao.p.resources?.includes(`ceramic://*?model=${MODEL_STREAM_ID}`)
+    if (cacao.p.resources?.length != 1 || !hasModelResource) {
```

Comment:
```suggestion
    if (!hasModelResource) {
```
Seems incorrect to ensure only one resource is allowed in the CACAO, no? The CACAO could be used for other resources as well.